linera-base/src/crypto/mod.rs

```diff
@@ -28,6 +30,36 @@ pub type ValidatorSignature = secp256k1::Secp256k1Signature;
 /// The key pair of a validator.
 pub type ValidatorKeypair = secp256k1::Secp256k1KeyPair;
 
+/// Signature scheme used for the public key.
+#[derive(Serialize, Deserialize, Debug, Copy, Clone, Eq, PartialEq)]
+pub enum SignatureScheme {
+    /// Ed25519
+    Ed25519,
+    /// secp256k1
+    Secp256k1,
+}
+
+impl SignatureScheme {
+    /// Returns the flag of the signature scheme.
+    pub fn flag(&self) -> u8 {
+        match self {
+            SignatureScheme::Ed25519 => 0x00,
+            SignatureScheme::Secp256k1 => 0x01,
+        }
```

Comment:
Would it make sense to use `#[repr(u8)]` so we could just use `scheme as u8` as the flag?
(And should it be "tag" rather than "flag"?)